libraries/lib-wave-track/WaveTrack.cpp

```diff
@@ -213,19 +261,93 @@ void WaveTrack::Interval::StretchRightTo(double t)
       GetClip(channel)->StretchRightTo(t);
 }
 
-void WaveTrack::Interval::ApplyStretchRatio(
-   const std::function<void(double)>& reportProgress)
+WaveTrack::IntervalHolder WaveTrack::Interval::GetStretchRenderedCopy(
+   const std::function<void(double)>& reportProgress, const ChannelGroup& group,
+   const SampleBlockFactoryPtr& factory)
```

Comment:
This replacement for `WaveClip::ApplyStretchRatio` loses a few good things:
- Early quick exit when ratio is already 1
- pre- and post-conditions, stated in header comments, and asserted in the function